### PR TITLE
Rationalise logging in `receive_keys_claim_response`

### PR DESCRIPTION
**Pull request by richvdh**

### CODE REVIEW

File: crates/matrix-sdk-crypto/src/session_manager/sessions.rs [hunk 1]
```diff
@@ -414,18 +414,23 @@ impl SessionManager {
...
+                        .or_default()
+                        .insert((*device_id).clone());
+                }
+
                 warn!(
-                    ?missing_devices,
-                    "Tried to create a new sessions, but the signed one-time key was missing for some devices",
+                    ?missing_devices_by_user,
+                    "Tried to create new Olm sessions, but the signed one-time key was missing for some devices",
                 );
```
Comment (richvdh) [inline crates/matrix-sdk-crypto/src/session_manager/sessions.rs:428]: Logging the missing devices in this way means that we don't have to repeat the user id for each device, and generally makes it (a bit) easier to read.

File: crates/matrix-sdk-crypto/src/session_manager/sessions.rs [hunk 2]
```diff
@@ -414,18 +414,23 @@ impl SessionManager {
                 .collect();
 
             if !missing_devices.is_empty() {
+                let mut missing_devices_by_user: BTreeMap<_, BTreeSet<_>> = BTreeMap::new();
+                for (user_id, device_id) in missing_devices.into_iter() {
```
Comment (jplatte) [inline crates/matrix-sdk-crypto/src/session_manager/sessions.rs:418]:
```suggestion
                for (user_id, device_id) in missing_devices {
```

File: crates/matrix-sdk-crypto/src/session_manager/sessions.rs [hunk 3]
```diff
@@ -414,18 +414,23 @@ impl SessionManager {
...
+                    missing_devices_by_user
+                        .entry(user_id)
+                        .or_default()
+                        .insert((*device_id).clone());
```
Comment (jplatte) [inline crates/matrix-sdk-crypto/src/session_manager/sessions.rs:422]: What's the type of `device_id` here that makes an explicit dereference necessary?

Comment (richvdh) [reply crates/matrix-sdk-crypto/src/session_manager/sessions.rs:422]: it is `&&OwnedDeviceId`. 

Note that this explicit deref is not new; it's just moved. It was suggested by poljar in [this thread](https://github.com/matrix-org/matrix-rust-sdk/pull/2805#discussion_r1387901456).

**Review by jplatte — APPROVED** [review]

**Pull request merged**